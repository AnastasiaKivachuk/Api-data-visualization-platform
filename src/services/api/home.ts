import { gql } from '@apollo/client';
import client from '../../../apollo-client';

export const getHome = () => client.query({
  query: gql`
        query HomeScheme {
            company {
                ceo
                employees
                founded
                founder
                links {
                    elon_twitter
                    website
                }
                name
                summary
            }
            ships {
                id
                image
                model
                name
                url
                year_built
            }
            rockets {
                wikipedia
                name
                id
                description
            }
        }
    `,
});
