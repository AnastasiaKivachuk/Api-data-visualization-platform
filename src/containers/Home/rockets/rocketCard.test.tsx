import {shallow} from "enzyme";
import React from "react";

import Rockets from "./index";
import {RocketsInterface} from "../../../interfaces/home.interface";

const MockRocketsData: RocketsInterface[] = [{
    description: "The Falcon 1 was an expendable launch system privately developed and manufactured by SpaceX during 2006-2009. On 28 September 2008, Falcon 1 became the first privately-developed liquid-fuel launch vehicle to go into orbit around the Earth.",
    id: "falcon1",
    name: "Falcon 1",
    wikipedia: "https://en.wikipedia.org/wiki/Falcon_1"
}];

const setUp = (props: {rockets: RocketsInterface[]} | null) => shallow(<Rockets {...props} />);

describe("Rockets component", () => {

    it("should render Rockets component", () => {
        const component = setUp({rockets: MockRocketsData})
        expect(component).toMatchSnapshot();
    })

    it("check all elements with props", () => {
        const component = setUp({rockets: MockRocketsData})
        expect(component.find('ForwardRef(Typography)').text()).toEqual('Rockets');
        expect(component.find('RocketCard').length).toBe(1)
    })

    it("check all elements with defaultProps", () => {
        const component = setUp(null)
        expect(component.find('RocketCard').length).toBe(0)
    })
})
