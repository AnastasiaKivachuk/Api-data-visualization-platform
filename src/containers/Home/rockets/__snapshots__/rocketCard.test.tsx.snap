// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Rockets component should render Rockets component 1`] = `
<Styled(div)>
  <ForwardRef(Typography)
    gutterBottom={true}
    textAlign="center"
    variant="h4"
  >
    Rockets
  </ForwardRef(Typography)>
  <ForwardRef(Grid)
    container={true}
    gap="20px"
    justifyContent="center"
  >
    <RocketCard
      description="The Falcon 1 was an expendable launch system privately developed and manufactured by SpaceX during 2006-2009. On 28 September 2008, Falcon 1 became the first privately-developed liquid-fuel launch vehicle to go into orbit around the Earth."
      id="falcon1"
      key="falcon1"
      name="Falcon 1"
      wikipedia="https://en.wikipedia.org/wiki/Falcon_1"
    />
  </ForwardRef(Grid)>
</Styled(div)>
`;
